topic query generation (#32)
MrLongbottom/P9-Rapport: sections/method/query_handling.tex

We will describe each step in the following sections.

\subsection{Query Generation}
We use two techniques for generating the queries used for evaluating our search algorithm, namely document-queries and topic-queries.

> The idea behind these is to evaluate the generality and specificity of the \gls{lda} model.

Reviewer: ```suggestion
The idea behind these two methods is is to evaluate the specificity and generality of the \gls{lda} model, respectively.
```

The construction of the queries is described in the following.

\subsubsection{Document query}

> To construct a document query, we randomly select a document from the corpus and a length of the query which can be between 1-4 words.

Reviewer: ```suggestion
To construct a document query, we randomly select a document from the corpus and query length $ql$ between $ql_{min}$ and $ql_{max}$.
```

From this selection, we apply \gls{tf-idf} on the selected document to find the most informative words.

> Our query is then the highest ranked word within the given document.

Reviewer: ```suggestion
Our query is then the highest $ql$ ranked words within the given document.
```

> The goal is to check whether or not the model is able to find a specific document based on a search query.

Reviewer: ```suggestion
The goal is to check whether or not the model is able to find a specific document based on a search query generated from that document.
```

\subsubsection{Topic query} 
The topic query is slightly different in comparison to document query in that it focuses on the topics rather than a specific document.

> To construct a topic query, firstly we select a random topic $t$ and a length of a query $l_q$ to be between 1-4.

Reviewer: ```suggestion
To construct a topic query, firstly we select a random topic $t$ and a length of a query length $ql$ between $ql_{min}$ and $ql_{max}$.
```

> We then sample $l_q$ document from on the document-topic distribution $\theta_t$ provided by the \gls{lda} model. 

Reviewer: ```suggestion
We then sample $ql$ documents based on the document-topic distribution $\theta_t$ provided by the \gls{lda} model.
Though $theta$ is normally one topic distribution for each document, we create document distributions for each topic by row normalizing the matrix. 
```

When sampling these documents, we only sample within the document-topic distribution where the topic is set to $t$.
From these documents, we extract one word from each document and add these to the query based on the \gls{tf-idf} score.

> If an extracted word is in the query already, we skip it and add the second highest \gls{tf-idf} score to the query.

Reviewer: ```suggestion
If an extracted word is in the query already, we skip it and add the next highest \gls{tf-idf} score to the query.
```
er i tvivl om det er 'an extracted word' eller 'a extracted word'

In that way, we ensure to get unique words in the query.

> 

Reviewer: vi mangler lidt en sætning ligesom slutningen af document query generation hvor vi forklarer hvad idéen/målet er.

Author: True

Author: ```suggestion
The goal, of generating this query, is to check whether the model can find a similar document based on a search query generated from that topic distribution.
```

We will describe each step in the following sections.

\subsection{Query Generation}

> We use two techniques for generating the queries used for evaluating our search algorithm, namely document-queries and topic-queries.

Reviewer 2: ```suggestion
We use two techniques for generating the queries used for evaluating our search algorithm, namely document queries and topic queries.
```

The idea behind these two methods is to evaluate the specificity and generality of the \gls{lda} model, respectively.
The construction of the queries is described in the following.

\subsubsection{Document query}
To construct a document query, we randomly select a document from the corpus and a query length $ql$ between $ql_{min}$ and $ql_{max}$.
From this selection, we apply \gls{tf-idf} on the selected document to find the most informative words.

> Our query is then the highest $ql$ ranked words within the given document.

Reviewer 2: ```suggestion
Our query is then the $ql$ highest ranked words within the given document.
```

The goal is to check whether or not the model is able to find a specific document based on a search query generated from that document.

\subsubsection{Topic query} 
The topic query is slightly different in comparison to document query in that it focuses on the topics rather than a specific document.
To construct a topic query, firstly we select a random topic $t$ and a query length $ql$ between $ql_{min}$ and $ql_{max}$.
We then sample $ql$ documents based on the document-topic distribution $\theta_t$ provided by the \gls{lda} model.
Though $theta$ is normally one topic distribution for each document, we create document distributions for each topic by row normalizing the matrix. 
When sampling these documents, we only sample within the document-topic distribution where the topic is set to $t$.

> From these documents, we extract one word from each document and add these to the query based on the \gls{tf-idf} score.

Reviewer 2: ```suggestion
From these sampled documents, we extract one word from each document, based on the \gls{tf-idf} score, and add these to the query.
```

> If an extracted word is in the query already, we skip it and add the next highest \gls{tf-idf} score to the query.

Reviewer 2: ```suggestion
If an extracted word is in the query already, we skip it and add the next highest \gls{tf-idf} scored word to the query.
```
